core/src/main/java/io/opencensus/tags/TagContextBuilder.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.tags;
+
+import io.opencensus.tags.TagKey.TagKeyBoolean;
+import io.opencensus.tags.TagKey.TagKeyLong;
+import io.opencensus.tags.TagKey.TagKeyString;
+import javax.annotation.concurrent.Immutable;
+
+/** Builder for the {@link TagContext} class. */
+public abstract class TagContextBuilder {
+  private static final TagContextBuilder NOOP_TAG_CONTEXT_BUILDER = new NoopTagContextBuilder();
+
+  /**
+   * Adds the key/value pair regardless of whether the key is present.
+   *
+   * @param key the {@code TagKey} which will be set.
+   * @param value the value to set for the given key.
+   * @return this
+   * @throws IllegalArgumentException if either argument is null.
+   */
+  public abstract TagContextBuilder set(TagKeyString key, TagValueString value);
+
+  /**
+   * Adds the key/value pair regardless of whether the key is present.
+   *
+   * @param key the {@code TagKey} which will be set.
+   * @param value the value to set for the given key.
+   * @return this
+   * @throws IllegalArgumentException if the key is null.
+   */
+  // TODO(sebright): Make this public once we support types other than String.
+  protected abstract TagContextBuilder set(TagKeyLong key, long value);
+
+  /**
+   * Adds the key/value pair regardless of whether the key is present.
+   *
+   * @param key the {@code TagKey} which will be set.
+   * @param value the value to set for the given key.
+   * @return this
+   * @throws IllegalArgumentException if the key is null.
+   */
+  // TODO(sebright): Make this public once we support types other than String.
+  protected abstract TagContextBuilder set(TagKeyBoolean key, boolean value);
+
+  /**
+   * Removes the key if it exists.
```

Comment:
Nit: This style of javadoc is burdensome to read as it just restates the method in english. Id prefer we make docs lines and comments count, as useless comments draw the same attention as useful ones.